Fixes Issue#3178 : Organisation Settings Page UI according to Figma

AceHunterr opened the pull request:
**What kind of change does this PR introduce?**
UI Bugs Fix

**Issue Number:**
Fixes #3178 

**Snapshots/Videos:**

Here is the video for the setting UI changes made:

https://github.com/user-attachments/assets/32a37c4c-5969-489e-8928-d4a3391f1bca

<video controls width="600">
  <source src="https://github.com/user-attachments/assets/32a37c4c-5969-489e-8928-d4a3391f1bca" type="video/mp4">
  Your browser does not support the video tag.
</video>

1. Change the green colored UI to blue and grey according to the Figma
2. Fixed the Setting Tab and fixed the styling for the tab switching
3. Added the reset button and it's functionality which was missing according to the figma.
4. Fixed delete card UI and the switching button from green to blue and also reducing their width and adjusting flex.
5. Added subtle animation between switching pages to make the UI look a lot cleaner.
6. Refactored form input borders when focused and active from green to blue.
7. Also the colour of the save and other buttons which was mentioned in the Figma but not present in the code... added them and used them according to the Figma.
8. Updated the Languages for the "changesReset" variable in the public directory.


Also added a file for testing for the GeneralSettings.jsx and wrote those test in vitest framework and ensured a 100% coverage.

![entire_coverage](https://github.com/user-attachments/assets/19571ba6-9498-4bb7-bb7a-f3fdbcbf9f03)
![passed_tests](https://github.com/user-attachments/assets/c9b88148-cb20-4b04-87df-134264dd9c1e)


There was this issue coming with the Agenda Item Categories Component where it was not loading properly and I tried different ways to make it work but it was not working showing "Response Not Successful" asking on slack as well... I checked the UI via the code and it was good so far following the Figma but if this issue gets fixed it would be clear for the fix.


**If relevant, did you update the documentation?**
No

**Summary**

**Does this PR introduce a breaking change?**
No

## Checklist

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

<!--Yes or No-->

AceHunterr commented:
There are certain components that are present in the code but are not present in the figma... for example:

The change language component:
![1](https://github.com/user-attachments/assets/4db53213-e790-4646-98a9-f62711b98e77)

and the custom field component:
![2](https://github.com/user-attachments/assets/01e06a32-2891-4756-84bb-8c18105bf817)

.... shall I simply remove them?

And also for some reason my agenda items tab is showing error 404.... not sure whether this is an issue or happening with me only... I asked on slack as well and I tried fixing it too but still getting this:

![3](https://github.com/user-attachments/assets/8c24797e-6381-4de2-a714-7dcb0f1e02db)


Also I see that the form has an extra inline border that I will remove right away and also will get the whole code coverage in the next commit.

Rest about the colour scheme I checked all the colours and shades using the color picker in figma and designed the elements accordingly.
Is there anything else that I am missing with respect to the Figma design to work upon?

palisadoes commented:
Doesn't match figma

palisadoes reviewed (changes requested):
1. Doesn't match figma
    ![image](https://github.com/user-attachments/assets/9522a9c9-6cd0-42b5-8492-07f3fa382357)
1. See comment
    ![image](https://github.com/user-attachments/assets/dc8d023b-3a68-4427-b490-9d36f2146930)
1. More figma
    ![image](https://github.com/user-attachments/assets/2346c718-b9ac-4bae-8103-724bcdafd7ae)
1. Does not match the component CSS of this PR
    1. https://github.com/PalisadoesFoundation/talawa-admin/pull/2984
        ![image](https://github.com/user-attachments/assets/c96a9d7f-8d45-4f60-ae5b-1198d4c65ee5)
        ![image](https://github.com/user-attachments/assets/d31b63a5-b873-45a0-b545-a56e239653c7)
        ![image](https://github.com/user-attachments/assets/8ccf2a01-f23a-49d5-999f-5af98b8deee5)

1. Button colors don't match Figma
    ![image](https://github.com/user-attachments/assets/3d05c8f4-e942-4a51-9025-b4b7c7484313)
1. More
    ![image](https://github.com/user-attachments/assets/7d7867dd-c902-4516-978e-48b5a20736b0)
1. More
    ![image](https://github.com/user-attachments/assets/fc7f4df8-0eb5-40f9-b4ba-28e6a221c56c)
1. More
    ![image](https://github.com/user-attachments/assets/c14090b1-066a-41cf-bf93-11ea27299afc)
1. More
    ![image](https://github.com/user-attachments/assets/de967f7b-dc22-48c6-a7f0-0e44b3215b67)
1. More
    ![image](https://github.com/user-attachments/assets/39bec48c-0190-4d18-81e6-e6f130e2400b)
1. More
    1. This is not in figma
        ![image](https://github.com/user-attachments/assets/bb3f1363-e571-4f57-819f-4fa506adb38c)
    1. This is:
        ![image](https://github.com/user-attachments/assets/b48d91e6-8dc6-4f9f-bb37-f27f5abd47f1)
1. More
    ![image](https://github.com/user-attachments/assets/6058eb47-3faf-4bba-ae93-0210a011171a)
1. More
    ![image](https://github.com/user-attachments/assets/47227bf4-9861-49c8-a4c5-e07f3f6e6d21)

palisadoes reviewed (changes requested):
1. Use the user profile page when in doubt. 
    ![image](https://github.com/user-attachments/assets/922d10c9-39d2-45f1-8f33-afd9c64c0d50)
1. Consider changing what's below to match the suggestions in the image above
![image](https://github.com/user-attachments/assets/9962d1f5-e0ba-4b0c-ad6d-18ddbd263d53)
1.

palisadoes reviewed (changes requested):
![image](https://github.com/user-attachments/assets/0d5a166c-3783-4fd5-b155-bd32a561b7f5)

palisadoes commented:
Don't delete any elements from the code. The features are required

palisadoes commented:
I've made a lot of suggested changes. The figma doesn't cover absolutely all scenarios. You'll need to look to similar screens for inspiration. Some figma colors are not consistent, there are a few inappropriate blue button colors and some backgrounds are not consistent

palisadoes commented:
The end product will look much better

AceHunterr commented:
Okay sure I'll work up these changes.

palisadoes commented:
Please make a minor commit. The failing test should be fixed with that

palisadoes commented on src/components/OrgSettings/ActionItemCategories/CategoryModal.tsx:
```diff
@@ -171,6 +171,16 @@ const CategoryModal: FC<InterfaceActionItemCategoryModal> = ({
+              sx={{
+                '& .MuiOutlinedInput-root': {
+                  '&:hover fieldset': {
+                    borderColor: '#a8c7fa',
```
1. Don't use hard coded colors
1. Make this file use CSS exclusively from `src/style/app.module.css`, with the exception of package based CSS.
1. Delete the CSS file it currently uses.
1. Use no CSS in `assets/css/app.css`
1. This will help with future attempts to having Dark Mode
1. Use a variable from `src/style/app.module.css` .
1. Use a variable name that describes the function of the color and not the color itself. For example `--search-button-bg` is much better than `--light-blue` 

- https://github.com/PalisadoesFoundation/talawa-admin/issues/3296

palisadoes commented on src/components/OrgSettings/ActionItemCategories/CategoryModal.tsx:
```diff
@@ -171,6 +171,16 @@ const CategoryModal: FC<InterfaceActionItemCategoryModal> = ({
+                    borderColor: '#a8c7fa',
+                  },
+                  '&.Mui-focused fieldset': {
+                    borderColor: '#a8c7fa',
```
1. Don't use hard coded colors
1. Make this file use CSS exclusively from `src/style/app.module.css`, with the exception of package based CSS.
1. Delete the CSS file it currently uses.
1. Use no CSS in `assets/css/app.css`
1. This will help with future attempts to having Dark Mode
1. Use a variable from `src/style/app.module.css` .
1. Use a variable name that describes the function of the color and not the color itself. For example `--search-button-bg` is much better than `--light-blue` 

- https://github.com/PalisadoesFoundation/talawa-admin/issues/3296

palisadoes commented on src/components/OrgSettings/General/DeleteOrg/DeleteOrg.tsx:
```diff
@@ -121,7 +123,7 @@ function deleteOrg(): JSX.Element {
           data-testid="orgDeleteModal"
         >
-          <Modal.Header className="bg-primary" closeButton>
+          <Modal.Header style={{ backgroundColor: '#a8c7fa' }} closeButton>
```
1. Don't use hard coded colors
1. Make this file use CSS exclusively from `src/style/app.module.css`, with the exception of package based CSS.
1. Delete the CSS file it currently uses.
1. Use no CSS in `assets/css/app.css`
1. This will help with future attempts to having Dark Mode
1. Use a variable from `src/style/app.module.css` .
1. Use a variable name that describes the function of the color and not the color itself. For example `--search-button-bg` is much better than `--light-blue` 

- https://github.com/PalisadoesFoundation/talawa-admin/issues/3296

palisadoes reviewed (changes requested):
1. See comments
2. Make code rabbit approve your changes
3. Add tests to make the files you edited achieve as close to 100% code coverage as possible

AceHunterr commented:
> Please make a minor commit. The failing test should be fixed with that

Actually I am really stuck with the Image Picker part... I am not being able to get the preview of the images on the organisation and I am trying to fetch the stored ones but is not able to locate that.... I have made the rest of the UI changes... I'll just comment the image picker part for now and keep working on that meanwhile I just commit the rest of the changes.

palisadoes commented:
Please post a video of what you have so far

AceHunterr commented:
Here is the video for the setting UI changes made:

https://github.com/user-attachments/assets/e0b7b058-33b5-498e-90e6-dc73b8946256

<video controls width="600">
  <source src="https://github.com/user-attachments/assets/e0b7b058-33b5-498e-90e6-dc73b8946256" type="video/mp4">
  Your browser does not support the video tag.
</video>




Changes Made:

1. Removed the hover color changes effects from all the elements.... used grey shadows for the selected fields,buttons,dropdowns etc.
2. Fixed the form UI according to the Figma and the screenshots ... removed lines,bold texts and the input field focused outline.
3. Modified the Category and Delete Modal and shifted the top right close button to bottom right close button.
4. Fixed button colours and used a standard color given the color picker from the figma ... did this for every component.
5. Fixed the MUI datagrid by removing bottom row borders , green svgs and texts for active or disabled and table header.

Changes left to made:
1. Create an image picker element and place it under "Profile Details".
2. Club action buttons into one section under the "Profile Details"

AceHunterr commented on src/components/OrgSettings/General/OrgUpdate/ImagePicker.tsx:
```diff
@@ -0,0 +1,62 @@
+    }
+  };
+
+  return (
+    <div className={styles.imagePicker}>
+      <div className={styles.imagePickerContainer}>
+        <img
+          src={previewImage || '/path/to/default-image.jpg'} // Default placeholder image.
+          alt="Preview"
+          className={styles.previewImage}
+        />
+        <div className={styles.overlay}>
+          <input
+            type="file"
+            accept="image/*"
+            className={styles.fileInput}
+            onChange={handleImageChange}
+          />
+          <FaCamera className={styles.cameraIcon} />
+        </div>
+      </div>
+    </div>
+  );
```
Resolving this conversation for time being because this component is not complete and not being used anywhere

AceHunterr commented:
@palisadoes may I upload a video of what exactly I am facing issue with for the Image Picker and someone may help here?

AceHunterr commented on src/components/OrgSettings/General/OrgUpdate/ImagePicker.tsx:
```diff
@@ -0,0 +1,68 @@
+/**
+ * Props for the `ImagePicker` component.
+ */
+interface InterfaceImagePickerProps {
+  defaultImage?: string; // The default image to display in the preview.
+  onImageSelect: (base64Image: string) => void; // Callback for when an image is selected.
+  defaultPlaceholderImage: string;
+}
```
Resolving this conversation for now since the component is not complete and not being used currently will have to update later.

AceHunterr commented:
https://github.com/user-attachments/assets/73e4753c-2c7e-4f72-84d6-6a9324e818a3

<video controls width="600">
  <source src="https://github.com/user-attachments/assets/73e4753c-2c7e-4f72-84d6-6a9324e818a3" type="video/mp4">
  Your browser does not support the video tag.
</video>

What is happening is that when I am trying to upload an image for organisation I am getting no image found on the console and when I am checking the code particularly here:

![image](https://github.com/user-attachments/assets/47141a92-7316-4145-a636-b5740f1e4cca)

we are storing the image as the base64 encoded string... but it is not loading... As I shown in the video above... the current file uploader we are using and then the imagePicker that I am trying to implement(UI not present much.. was trying to get the functionality first)

AceHunterr commented:
@coderabbitai full review

AceHunterr commented:
@coderabbitai full review

AceHunterr commented on src/assets/css/app.css:
```diff
@@ -2456,6 +2456,7 @@ progress {
   border: 1px solid var(--bs-border-color);
   appearance: none;
   border-radius: var(--bs-border-radius);
+  box-shadow: none;
```
+ outline: 2px solid var(--subtle-blue-grey);
+ outline-offset: 2px;

By applying these 2 styles we are violating the Figma code of no color change on hover and focus... can we simply do this one thing where we have a very light border on focus and hover to maintain the subtleness?

palisadoes commented:
1. Please ask for assistance in the #talawa-admin slack channel.
2. You also have numerous broken images in the video that could be caused for similar reasons.

AceHunterr commented:
Okay I'll create a new branch for the changes
